src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnectionImpl.java

```diff
@@ -971,13 +971,17 @@ public Integer[] addNotificationListeners(ObjectName[] names,
         if (names == null || filters == null) {
             throw new IllegalArgumentException("Got null arguments.");
         }
+        // Accept an array of delegationSubjects from e.g. earlier JDKs,
+        // but throw if it contains any non-null values.
         if (delegationSubjects != null) {
-            throw new UnsupportedOperationException("Subject Delegation has been removed.");
+            for (Subject s: delegationSubjects) {
+                if (s != null) {
+                    throw new UnsupportedOperationException("Subject Delegation has been removed.");
```

Comment:
Q1: Would it make sense to provide more specific about the failing `delegationSubject` element?
Q2: Does this fix needs a CSR?

Reply:
Q1 We have a couple of places where we use that text, when a non-null Subject is given.  It's quite generic and that is a good fit.  Your app would have to go to a significant effort to use the Subject Delegation feature, and we aren't aware of any apps that use it.  If JConsole were using the feature, you would know as you would have had to specifically configure it somehow (as well as on the remote end with SM and policies), so I really think the generic message is explicit enough.

Q2 Yes, Daniel very quickly added the CSR flag.  I was presuming that "paperwork" would need doing, although yes as spec changes go, it is very very minor.